### PR TITLE
Language gets changed on uploading

### PR DESCRIPTION
**Pull request by AmeyaJain-25**
Issue #111 

Issue is resolved.
The language changes according to the file extension

### CODE REVIEW

File: frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js [hunk 1]
```diff
@@ -152,6 +153,9 @@ const SyntaxEditor = (props) => {
         }
 
         handleChange(e.target.result);
+        const fileName = file.name;
+        const ext = `.${fileName.split(".")[1]}`;
```
[the remark this answers is not in the document]
Comment (AmeyaJain-25) [reply frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js:157]: Yes, perfect.
I'll do that

File: frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ const useStyles = makeStyles((mutheme) => ({
 }));
 
 const validExtensions = [".c", ".cpp", ".java", ".js", ".ts", ".clj", ".cljs", ".cs", ".cbl", ".cob", ".cpy", ".erl", ".hrl", ".go", ".py", ".f90", ".f95", ".f03", ".txt", ".groovy", ".gvy", ".gy", ".gsh", ".kt", ".kts", ".ktm", ".php", ".r", ".rb", ".sql", ".swift"];
+const langExtensionDict = {".c": "C", ".cpp": "C++", ".java": "JAVA", ".js": "JavaScript", ".ts": "TypeScript", ".clj": "Closure", ".cljs": "Closure", ".cs": "C#", ".cbl": "COBOL", ".cob": "COBOL", ".cpy": "COBOL", ".erl": "Erlang", ".hrl": "Erlang", ".go": "Go", ".py": "Python", ".f90": "FortRan", ".f95": "FortRan", ".f03": "FortRan", ".txt": "", ".groovy": "Groovy", ".gvy": "Groovy", ".gy": "Groovy", ".gshgsh": "Groovy", ".kt": "Kotlin", ".kts": "Kotlin", ".ktm": "Kotlin", ".php": "PHP", ".r": "R", ".rb": "Ruby", ".sql": "SQL", ".swift": "Swift" };
```
[the remark this answers is not in the document]
Comment (AmeyaJain-25) [reply frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js:62]: Yes. I'll try that

**Comment by AmeyaJain-25** [comment]
@Vishv07 

Now, the file extension is taking the last paramters after ( . )
Tried with files like app.test.js, etc. Working fine.

Also,
As there are multiple extensions for particular languages like eg: kotlin has .kt, .kts, .ktm.
Hence we cannot make changes to existing langOptions or other arrays.

So, I created that langExtensionDict object.

And the valid extension array is removed and have used from the keys of langExtensionDict as an array for validExtensions.

**Review by kothariji — APPROVED** [review]
@AmeyaJain-25 Thanks for the contributions 🚀

Keep Contributing 🔥

**Pull request merged**